numpy/f2py/src/fortranobject.c

```diff
@@ -986,47 +1071,56 @@ int check_and_fix_dimensions(const PyArrayObject* arr, const int rank, npy_intp
                          new_size, arr_size);
             return 1;
         }
-    } else { /* [[1,2]] -> [[1],[2]] */
-        int i,j;
+    }
+    else { /* [[1,2]] -> [[1],[2]] */
+        int i, j;
         npy_intp d;
         int effrank;
         npy_intp size;
-        for (i=0,effrank=0;i<PyArray_NDIM(arr);++i)
-            if (PyArray_DIM(arr,i)>1) ++effrank;
-        if (dims[rank-1]>=0)
-            if (effrank>rank) {
+        for (i = 0, effrank = 0; i < PyArray_NDIM(arr); ++i)
+            if (PyArray_DIM(arr, i) > 1)
+                ++effrank;
+        if (dims[rank - 1] >= 0)
+            if (effrank > rank) {
                 PyErr_Format(PyExc_ValueError,
                              "too many axes: %d (effrank=%d), "
                              "expected rank=%d\n",
                              PyArray_NDIM(arr), effrank, rank);
                 return 1;
             }
 
-        for (i=0,j=0;i<rank;++i) {
-            while (j<PyArray_NDIM(arr) && PyArray_DIM(arr,j)<2) ++j;
-            if (j>=PyArray_NDIM(arr)) d = 1;
-            else d = PyArray_DIM(arr,j++);
-            if (dims[i]>=0) {
-                if (d>1 && d!=dims[i]) {
-                    PyErr_Format(PyExc_ValueError,
-                                 "%d-th dimension must be fixed to %"
-                                 NPY_INTP_FMT " but got %" NPY_INTP_FMT
-                                 " (real index=%d)\n",
-                                 i, dims[i], d, j-1);
+        for (i = 0, j = 0; i < rank; ++i) {
+            while (j < PyArray_NDIM(arr) && PyArray_DIM(arr, j) < 2) ++j;
+            if (j >= PyArray_NDIM(arr))
+                d = 1;
+            else
+                d = PyArray_DIM(arr, j++);
+            if (dims[i] >= 0) {
+                if (d > 1 && d != dims[i]) {
+                    PyErr_Format(
+                            PyExc_ValueError,
+                            "%d-th dimension must be fixed to %" NPY_INTP_FMT
+                            " but got %" NPY_INTP_FMT " (real index=%d)\n",
+                            i, dims[i], d, j - 1);
                     return 1;
                 }
-                if (!dims[i]) dims[i] = 1;
-            } else
+                if (!dims[i])
+                    dims[i] = 1;
+            }
+            else
                 dims[i] = d;
         }
 
-        for (i=rank;i<PyArray_NDIM(arr);++i) { /* [[1,2],[3,4]] -> [1,2,3,4] */
-            while (j<PyArray_NDIM(arr) && PyArray_DIM(arr,j)<2) ++j;
-            if (j>=PyArray_NDIM(arr)) d = 1;
-            else d = PyArray_DIM(arr,j++);
-            dims[rank-1] *= d;
+        for (i = rank; i < PyArray_NDIM(arr);
+             ++i) { /* [[1,2],[3,4]] -> [1,2,3,4] */
+            while (j < PyArray_NDIM(arr) && PyArray_DIM(arr, j) < 2) ++j;
```

Comment:
I'm also surprised this is allowed.

Reply:
This can be controlled with the `ShortBlockStyle` option.

Reply:
The static analyzer isn't very smart, fixed this manually.

Reply:
That's just ugly. What if you put the trailing comment after the first `for` on the next line?

Reply:
Can you try adding:
```
ShortBlockStyle: SBS_Empty
```
And see if that helps some things?